app/javascript/Pages/SingleClass.jsx

```diff
@@ -0,0 +1,220 @@
+    switch (buttonState) {
+      case "pending":
+        //cancel request
+        setButtonState("loading");
+
+        axios
+          .put(`/api/match_request/${requestInfo.match_id}/`, {
+            status: "cancelled",
+            headers: window.defaultHeaders,
+          })
+          .then((response) => {
+            console.log(response.data);
+
+            setButtonState("available");
+          })
+          .catch((error) => {
+            if (error.response.data.error) {
+              alert(error.response.data.error);
+            }
+            setButtonState("pending");
+          });
+
+        break;
+
+      case "available":
+        //send request
+        setButtonState("loading");
+
+        axios
+          .post(`/api/classes/${props.class.id}/request/`, {
+            headers: window.defaultHeaders,
+          })
+          .then((response) => {
+            console.log(response.data);
+
+            requestInfo.match_id = response.data.request.id;
+            setButtonState("pending");
+          })
+          .catch((error) => {
+            if (error.response.data.error) {
+              alert(error.response.data.error);
+            }
+            setButtonState("available");
+          });
+
+        break;
+
+      default:
+        break;
+    }
+  };
+
+  //button state can be 'available', 'pending', 'accepted' or 'loading'
+  const [buttonState, setButtonState] = useState(
+    props.request.status ? props.request.status : "available"
+  );
+
+  const classes_url_category = () => {
+    return `/classes?category_id=${props.class.skill.category.id}`;
+  };
+
+  const classes_url_skill = () => {
+    return `/classes?category_id=${props.class.skill.category.id}&skill_id=${props.class.skill.id}`;
+  };
+
+  return (
+    <Layout current_user={props.current_user}>
+      <div className="path-wrapper">
+        <Link rel="stylesheet" href="/home" className="path">
+          <img
+            src={require("../images/logo_dark.svg")}
+            alt="logo"
+            className="path-logo"
+          />
+        </Link>
+        <p className="path">&nbsp;/&nbsp;</p>
+        <Link rel="stylesheet" href={classes_url_category()} className="path">
+          {props.class.skill.category.name}
+        </Link>
+        <p className="path">&nbsp;/&nbsp;</p>
+        <Link rel="stylesheet" href={classes_url_skill()} className="path">
+          {props.class.skill.name}
+        </Link>
+        <p className="path">&nbsp;/&nbsp;</p>
+        <Link rel="stylesheet" href="" className="path">
+          {props.class.teacher.name}
+        </Link>
+      </div>
+      <div
+        className="class-header"
+        style={{ "--colorCategory": `#${props.class.skill.category.color}` }}
+      >
+        <h1 className="class-header-title">{props.class.title}</h1>
+        <h2 className="class-header-subtitle">{props.class.teacher.name}</h2>
+      </div>
+      <div>
+        <ul className="class-menu">
+          <li
```

Comment:
**nitpick (non-blocking)**: when you have almost the same code repeated more than 2 times it's worth using a loop